### PR TITLE
First rough draft of BYOC docs

### PR DESCRIPTION
**Pull request by dchw**

### CODE REVIEW

File: docs/cloud/satellites/byoc/aws.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Installing BYOC in AWS
+
+This page documents the requirements and steps required to install BYOC Satellites in AWS.
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/aws.md:3]:
```suggestion
This page documents the requirements and steps required to install [BYOC](https://earthly.dev/pricing) Satellites in AWS.
```

Comment (brandonSc) [reply docs/cloud/satellites/byoc/aws.md:3]: Should we briefly re-iterate what BYOC is here in case a user lands here and is confused? (and doesn't click the link)

Comment (dchw) [reply docs/cloud/satellites/byoc/aws.md:3]: Why would we link to pricing?

File: docs/cloud/satellites/byoc/aws.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+## Requirements
+Before you begin to provision your BYOC configuration within AWS, there are a few requirements you need to make sure you meet first:
+
+### Networking
+Because every configuration is different, using BYOC within your organization will require you to configure your networking to match your use case. You'll need to ensure that:
+
+* Traffic from Earthly clients (build machines or otherwise) can reach the satellites directly. When installing the AWS CloudFormation template, a security group rule is created that should allow the correct ingress from the specified CIDR block.
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/aws.md:11]:
```suggestion
* Traffic from Earthly clients, including CI build runners and developer workstations, can reach the satellites directly. When installing the AWS CloudFormation template, a security group rule is created that should allow the correct ingress from the specified CIDR block.
```

File: docs/cloud/satellites/byoc/aws.md (hunk 1 above)
Comment (brandonSc) [reply docs/cloud/satellites/byoc/aws.md:3]: The pricing page has the BYOC plan and explains what it is. It might be useful if someone has landed on this page but hasn't seen the pricing page yet.

File: docs/cloud/satellites/byoc/aws.md (hunk 2 above)
Comment (brandonSc) [reply docs/cloud/satellites/byoc/aws.md:11]: Should we link to AWS docs for CloudFormation, security groups, etc, in case the user is not familiar with AWS?

File: docs/cloud/satellites/byoc/aws.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+* Traffic to any required resources (e.g. private repositories, the internet, etc) are allowed. When installing the AWS CloudFormation template, the default egress rule allows any outbound traffic.
+* Internal AWS DNS names must resolve to an address reachable on the network. This is because invocations of Earthly that reference the BYOC satellite will use the AWS internal DNS address to try and reach the satellite.
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/aws.md:13]:
```suggestion
* Internal AWS DNS names must resolve to an address reachable on the network. This is because invocations of Earthly that reference the BYOC satellite will use the AWS internal DNS address to connect to the satellite. 
```

File: docs/cloud/satellites/byoc/aws.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+Before you begin to provision your BYOC configuration within AWS, there are a few requirements you need to make sure you meet first:
+
+### Networking
+Because every configuration is different, using BYOC within your organization will require you to configure your networking to match your use case. You'll need to ensure that:
+
+* Traffic from Earthly clients (build machines or otherwise) can reach the satellites directly. When installing the AWS CloudFormation template, a security group rule is created that should allow the correct ingress from the specified CIDR block.
+* Traffic to any required resources (e.g. private repositories, the internet, etc) are allowed. When installing the AWS CloudFormation template, the default egress rule allows any outbound traffic.
+* Internal AWS DNS names must resolve to an address reachable on the network. This is because invocations of Earthly that reference the BYOC satellite will use the AWS internal DNS address to try and reach the satellite. 
+
+These can all be accomplished with most VPN technologies. We recommend and have direct experience with Tailscale. If you need help configuring other networking scenarios, please reach out to us!
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/aws.md:15]: Should we include links to our support email and community slack here?

File: docs/cloud/satellites/byoc/aws.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+### Cloud
+Configuring BYOC in your AWS account is as simple as applying a CloudFormation template, and reading its outputs. You'll need:
+
+* An account with permissions to create a new CloudFormation stack in AWS.
+* An account with permissions to describe an existing CloudFormation stack in AWS.
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/aws.md:21]: Should these be the _same_ account? I.e. is it `an account with permissions to create and describe CloudFormation stacks`?

File: docs/cloud/satellites/byoc/byoc.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
+# Bring Your Own Cloud
+
+"Bring Your Own Cloud" (BYOC) satellites **Experimental** are a hybrid between [self hosted](../self-hosted.md) and [earthly-managed](../../satellites.md) satellites. These satellites are managed by Earthly; just like our managed offering, but within your infrastructure. This blends the ease-of-use of managed satellites with the security benefits that self hosting can bring.
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/byoc.md:3]:
```suggestion
"Bring Your Own Cloud" (BYOC) satellites **Experimental** are a hybrid between [Self-Hosted](../self-hosted.md) and [Cloud](../../satellites.md) satellites. These satellites are managed by Earthly; just like our managed offering, but within your infrastructure. This blends the ease-of-use of managed satellites with the security benefits that self hosting can bring.
```

We tend to call these "Cloud" satellites in our pricing page and in other [doc pages](https://docs.earthly.dev/docs/remote-runners).

File: docs/cloud/satellites/byoc/byoc.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+*  earthly-cloud  2           Green  
+```
+The `*` indicates the default cloud that will be used when launching satellites within your organization, unless otherwise specified.
+
+
+{% hint style='info' %}
+Note that the `earthly-cloud` installation is a special cloud present in all organizations. Satellites within this cloud are managed within Earthly's cloud, by Earthly. You can change back to use Earthly-managed satellites at any time by running the `earthly cloud use earthly-cloud` installation.
+{% endhint %}
+
+To test-launch a new satellite within the cloud, run `earthly satellite launch --cloud <name> my-byoc-sat`. This will launch a new satellite using your cloud. Assuming that works, kick the tires by trying to run one of your builds on it!
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/byoc.md:64]:
```suggestion
To test-launch a new satellite within your new cloud, run `earthly satellite launch --cloud <name> my-byoc-sat`. This will launch a new satellite using your cloud. Assuming that works, kick the tires by trying to run one of your builds on it!
```

File: docs/cloud/satellites/byoc/tailscale.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Configuring Tailscale for BYOC
+
+Tailscale is a super-simple VPN that is easy to set up, and works well with BYOC satellites. This page documents the required configuration within Tailscale to enable BYOC.
+
+## Requirements
+
+* Configure a [subnet router](https://tailscale.com/kb/1019/subnets) to provide access to satellites for users on the VPN. This is required because satellites never join a VPN directly, and may change IP/DNS addresses frequently.
+* [Configure Restricted Nameservers (Split DNS)](https://tailscale.com/kb/1054/dns?q=split+dns#nameservers) to resolve custom DNS names, as required by your cloud provider.
+* If you are running Earthly from within a Kubernetes pod, or GHA runner; you may need to make use of the userspace networking mode.
```
Comment (brandonSc) [inline docs/cloud/satellites/byoc/tailscale.md:9]:
```suggestion
* If you are running Earthly from within a Kubernetes pod, or GHA runner; you may need to make use of the [userspace networking mode](https://tailscale.com/kb/1112/userspace-networking).
```

Comment (brandonSc) [reply docs/cloud/satellites/byoc/tailscale.md:9]: Should we advice them to use the official Github actions module from Tailscale in GHA?

Comment (brandonSc) [reply docs/cloud/satellites/byoc/tailscale.md:9]: We could also include full working examples here of using Tailscale from Github actions and Kuberentes, based on our own internal testing. I can help with the examples if you'd like, since it's all in my branch right now.

**Comment by brandonSc** [comment]
I left some minor comments, but I think the docs are great overall! I like all the screenshots, that should make it easy to follow along with, especially when not familiar with CloudFormation.

File: docs/cloud/satellites/byoc/tailscale.md (hunk 8 above)
Comment (dchw) [reply docs/cloud/satellites/byoc/tailscale.md:9]: I would like to avoid reproducing the documentation as much as possible. If its clear and step-by-step on Tailscale's side, we should keep it there. If there are Earthly-specific tweaks, then we should document those more?

File: docs/cloud/satellites/byoc/aws.md (hunk 5 above)
Comment (dchw) [reply docs/cloud/satellites/byoc/aws.md:21]: I was referencing user accounts not aws accounts

**Comment by brandonSc** [comment]
Should we update the table on this page? 
https://docs.earthly.dev/docs/remote-runners

File: docs/cloud/satellites/byoc/byoc.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+# Bring Your Own Cloud
+
+"Bring Your Own Cloud" (BYOC) satellites **Experimental** are a hybrid between [self hosted](../self-hosted.md) and [earthly-managed](../../satellites.md) satellites. These satellites are managed by Earthly; just like our managed offering, but within your infrastructure. This blends the ease-of-use of managed satellites with the security benefits that self hosting can bring.
+
+Here is how Earthly-managed satellites compare to BYOC:
```
Comment (vladaionescu) [inline docs/cloud/satellites/byoc/byoc.md:5]: The text is a bit too small when scanning quickly, and it helps to know what the table represents.

```suggestion
### Earthly BYOC Satellites vs Earthly Cloud Satellites
```

File: docs/cloud/satellites/byoc/byoc.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+Here is how Earthly-managed satellites compare to BYOC: 
+
+| Similarities                                                                               | Differences                                                           |
+|--------------------------------------------------------------------------------------------|-----------------------------------------------------------------------|
+| ✅ The ability to use `earthly satellite` commands to provision and decommission satellites | ❌ It is up to you to get traffic to and from the satellite            |
+| ✅ Earthly-provided monitoring and reliability                                              | ❌ Earthly cannot access the machines for debugging or troubleshooting |
+| ✅ Automatic updates                                                                        |                                                                       |
+| ✅ Automatic sleep/wake                                                                     |                                                                       |
+
+Here is how Self-hosted satellites compare to BYOC:
```
Comment (vladaionescu) [inline docs/cloud/satellites/byoc/byoc.md:14]:
```suggestion
### Earthly BYOC Satellites vs Earthly Self-Hosted Satellites
```

File: docs/cloud/satellites/byoc/byoc.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+| Similarities                                                     | Differences                                                                            |
+|------------------------------------------------------------------|----------------------------------------------------------------------------------------|
+| ✅ The satellite lives in your infrastructure, next to your tools | ❌ You cannot manually provision satellites                                             |
+| ✅ You pay for the usage the satellite incurs                     | ❌ You must allow Earthly access to at least some portion of an AWS account you control |
+|                                                                  | ❌ You pay the cloud provider, not Earthly, for the compute usage                       |
```
Comment (vladaionescu) [inline docs/cloud/satellites/byoc/byoc.md:20]: The most important difference I would add here is that we manage the up-time of satellites, whereas with self-hosted, they manage the up-time.

File: docs/cloud/satellites/byoc/byoc.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+## Installation
+
+### Step 1: Configure Your Cloud Provider
+
+Right now, BYOC Satellites are only supported in AWS. See our [AWS CloudFormation](./aws.md) instructions to provision the required AWS resources.
```
Comment (vladaionescu) [inline docs/cloud/satellites/byoc/byoc.md:26]: Harder to skip if we start with the actual step as the first sentence.

```suggestion
Follow the instructions on the [AWS CloudFormation](./aws.md) page to provision the required AWS resources. Right now, BYOC Satellites are only supported in AWS.
```

File: docs/cloud/satellites/byoc/byoc.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+Note that the `earthly-cloud` installation is a special cloud present in all organizations. Satellites within this cloud are managed within Earthly's cloud, by Earthly. You can change back to use Earthly-managed satellites at any time by running the `earthly cloud use earthly-cloud` installation.
+{% endhint %}
+
+To test-launch a new satellite within the cloud, run `earthly satellite launch --cloud <name> my-byoc-sat`. This will launch a new satellite using your cloud. Assuming that works, kick the tires by trying to run one of your builds on it!
+
+If everything looks good, you can run `earthly cloud use <name>` to set this cloud to be the default for your organization. 
+
+{% hint style='warning' %}
+This setting is global for all users within the org. This prevents people from launching satellites in the wrong cloud, and accidentially disseminating information that shouldn't be.
+{% endhint %}
```
Comment (vladaionescu) [inline docs/cloud/satellites/byoc/byoc.md:70]: I would make this explicitly step 6, and again, put the command on its own line so that it stands out.

File: docs/cloud/satellites/byoc/byoc.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+| ✅ Automatic sleep/wake                                                                     |                                                                       |
+
+### Earthly BYOC Satellites vs Earthly Self-Hosted Satellites
+
+| Similarities                                                     | Differences                                                                            |
+|------------------------------------------------------------------|----------------------------------------------------------------------------------------|
+| ✅ The satellite lives in your infrastructure, next to your tools | ❌ You cannot manually provision satellites                                             |
+| ✅ You pay for the usage the satellite incurs                     | ❌ You must allow Earthly access to at least some portion of an AWS account you control |
+|                                                                  | ❌ You pay the cloud provider, not Earthly, for the compute usage                       |
+|                                                                  | ❌ You are responsible for the Satellites monitoring and reliability                    |
```
Comment (vladaionescu) [inline docs/cloud/satellites/byoc/byoc.md:23]: Reading through these tables again, I'm realizing that it's pretty hard to follow along. e.g. the difference "you are responsible for monitoring" isn't clear whether that refers to BYOC or self-hosted.

I think that these tables could be better represented like this:

### BYOC vs Cloud

| | Earthly Cloud | Earthly BYOC |
|-------|-------|-------|
| Who is responsible for monitoring and reliability of the Satellite | ✅ Earthly | ✅ Earthly |
| Satellites are deployed within your internal network | ❌ No | ✅ Yes |
| Earthly Cloud and Earthly staff are prevented from accessing your internal network | ✅ N/A | ✅ Yes |
| How is compute billed | ✅ Zero-margin compute via Earthly | ✅ Supported by you via your cloud provider |
| Automatic updates | ✅ Yes | ✅ Yes |
| Auto-sleep to drastically reduce compute cost | ✅ Yes | ✅ Yes |
| Automatic management and GCing of cache volumes | ✅ Yes | ✅ Yes |
| Users can launch and remove satellites via the `earthly sat` CLI | ✅ Yes | ✅ Yes |
| Requires access to a set of limited AWS capabilities | ✅ No | ❌ Yes |


### BYOC vs Self-Hosted

| | Earthly Self-Hosted | Earthly BYOC |
|-------|-------|-------|
| Who is responsible for monitoring and reliability of the Satellite | ❌ You | ✅ Earthly |
| Satellites are deployed within your internal network | ✅ Yes | ✅ Yes |
| Earthly Cloud and Earthly staff are prevented from accessing your internal network | ✅ Yes | ✅ Yes |
| Compute is billed directly to you from your cloud provider | ✅ Yes | ✅ Yes |
| Automatic updates | ❌ No | ✅ Yes |
| Auto-sleep to drastically reduce compute cost | ❌ No | ✅ Yes |
| Automatic management and GCing of cache volumes | ❌ No | ✅ Yes |
| Users can launch and remove satellites via the `earthly sat` CLI | ❌ No | ✅ Yes |
| Requires access to a set of limited AWS capabilities | ✅ No | ❌ Yes |


Markdown for this:

```
### BYOC vs Cloud

| | Earthly Cloud | Earthly BYOC |
|-------|-------|-------|
| Who is responsible for monitoring and reliability of the Satellite | ✅ Earthly | ✅ Earthly |
| Satellites are deployed within your internal network | ❌ No | ✅ Yes |
| Earthly Cloud and Earthly staff are prevented from accessing your internal network | ✅ N/A | ✅ Yes |
| How is compute billed | ✅ Zero-margin compute via Earthly | ✅ Supported by you via your cloud provider |
| Automatic updates | ✅ Yes | ✅ Yes |
| Auto-sleep to drastically reduce compute cost | ✅ Yes | ✅ Yes |
| Automatic management and GCing of cache volumes | ✅ Yes | ✅ Yes |
| Users can launch and remove satellites via the `earthly sat` CLI | ✅ Yes | ✅ Yes |
| Requires access to a set of limited AWS capabilities | ✅ No | ❌ Yes |


### BYOC vs Self-Hosted

| | Earthly Self-Hosted | Earthly BYOC |
|-------|-------|-------|
| Who is responsible for monitoring and reliability of the Satellite | ❌ You | ✅ Earthly |
| Satellites are deployed within your internal network | ✅ Yes | ✅ Yes |
| Earthly Cloud and Earthly staff are prevented from accessing your internal network | ✅ Yes | ✅ Yes |
| Compute is billed directly to you from your cloud provider | ✅ Yes | ✅ Yes |
| Automatic updates | ❌ No | ✅ Yes |
| Auto-sleep to drastically reduce compute cost | ❌ No | ✅ Yes |
| Automatic management and GCing of cache volumes | ❌ No | ✅ Yes |
| Users can launch and remove satellites via the `earthly sat` CLI | ❌ No | ✅ Yes |
| Requires access to a set of limited AWS capabilities | ✅ No | ❌ Yes |
```

File: docs/cloud/satellites/byoc/byoc.md (hunk 6 above)
Comment (brandonSc) [reply docs/cloud/satellites/byoc/byoc.md:3]: I still think we should be consistent with how we name these products in our docs and pricing page. We haven't referred to Earthly-Managed in the other pages, so it might get confusing. 

I also think the term Earthly-managed could include BYOC because Earthly manages those clouds too.

**Review by vladaionescu — APPROVED** [review]
